include/onnxruntime/core/providers/cuda/cuda_provider_factory.h

```diff
@@ -4,6 +4,50 @@
 #include "onnxruntime_c_api.h"
```

Comment:
Yeah, this file is how our internal code interfaces with the Cuda provider (the ProviderInfo_CUDA interface is what the provider provides to us), but regular users should just use the C API instead of including this file for the OrtSessionOptionsAppendExecutionProvider_CUDA (ideally I'd just remove that function).

Reply:
Would it be better to move the interface functions to a separate file called cuda_provider_interfaces? Besides we can't remove OrtSessionOptionsAppendExecutionProvider_CUDA since the CUDA DLL is published and removing it would break ABI.